components/kyma-environment-broker/internal/process/deprovisioning/event_hub_step.go

```diff
@@ -0,0 +1,132 @@
+package deprovisioning
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/hyperscaler"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/hyperscaler/azure"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/process"
+	processazure "github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/process/azure"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/storage"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/storage/dberr"
+)
+
+type DeprovisionAzureEventHubStep struct {
+	OperationManager *process.DeprovisionOperationManager
+	instanceStorage  storage.Instances
+	processazure.EventHub
+}
+
+func NewDeprovisionAzureEventHubStep(os storage.Operations, is storage.Instances, hyperscalerProvider azure.HyperscalerProvider, accountProvider hyperscaler.AccountProvider, ctx context.Context) DeprovisionAzureEventHubStep {
+	return DeprovisionAzureEventHubStep{
+		OperationManager: process.NewDeprovisionOperationManager(os),
+		instanceStorage:  is,
+		EventHub: processazure.EventHub{
+			HyperscalerProvider: hyperscalerProvider,
+			AccountProvider:     accountProvider,
+			Context:             ctx,
+		},
+	}
+}
+
+var _ Step = (*DeprovisionAzureEventHubStep)(nil)
+
+func (s DeprovisionAzureEventHubStep) Name() string {
+	return "Deprovision Azure Event Hubs"
+}
+
+func (s DeprovisionAzureEventHubStep) Run(operation internal.DeprovisioningOperation, log logrus.FieldLogger) (internal.DeprovisioningOperation, time.Duration, error) {
+	hypType := hyperscaler.Azure
+
+	instance, err := s.instanceStorage.GetByID(operation.InstanceID)
```

Comment:
`getInstance()` is now split from `Run()`. But, for the rest, unfortunately, there is no gain of splitting because it's basically, a call then check for the error, which will be the same case if the call is split.